src/Chapter2.hs

```diff
@@ -348,8 +363,8 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf:: [a] -> [a]
+firstHalf l = (iterate init l)!! (length l - ((length l) `div` 2))
```

Comment:
Here as well you can use the `take` function instead of `iterate` 🙂 That is also will be more performant.